### ISSUE TITLE
IndexError: tuple index out of range

### ISSUE THREAD

**Issue by MinchoG** [sqlfluff/sqlfluff#668]
When trying to fix a 758 rows (another file is of 40 rows, so I doubt it is the length) of sql model that has both jinja and sql I get the quoted error below. I have an .sqlfile that has some macros defined and max_lline_lengthh = 160.
File starts like:
```
 {{
    config(
    materialized ='incremental',
    unique_key='SK_MIXPANEL_PK_EVENTS',
    schema = generate_schema_name('WAREHOUSE'),
    pre_hook='use warehouse WAREHOUSE_S_SIZE',
    post_hook='use warehouse WAREHOUSE_TRANSFORM_PROD',
  )
}}
with cte as (
select
   a,
   ...
from a
  left join b
{% if is_incremental() %}
    WHERE a.INGESTED_AT::date > (select max(INGESTED_AT::date) from {{this}} )
{% endif %}
and a few windows functions that repeat several times
```

```
sqlfluff lint models/5_WAREHOUSE/F_MIXPANEL_EVENTS.sql 
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/bin/sqlfluff", line 33, in <module>
    sys.exit(load_entry_point('sqlfluff==0.3.6', 'console_scripts', 'sqlfluff')())
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sqlfluff/cli/commands.py", line 172, in lint
    result = lnt.lint_paths(paths, verbosity=verbose, ignore_non_existent_files=False)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sqlfluff/linter.py", line 911, in lint_paths
    result.add(self.lint_path(path, verbosity=verbosity, fix=fix,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sqlfluff/linter.py", line 896, in lint_path
    self.lint_string(target_file.read(), fname=fname,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sqlfluff/linter.py", line 748, in lint_string
    lerrs, _, _, _ = crawler.crawl(parsed, dialect=config.get('dialect_obj'))
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sqlfluff/rules/base.py", line 268, in crawl
    dvs, raw_stack, child_fixes, memory = self.crawl(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sqlfluff/rules/base.py", line 233, in crawl
    res = self._eval(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sqlfluff/rules/std.py", line 53, in _eval
    if raw_stack[idx].type == 'whitespace':
IndexError: tuple index out of range
```

## Expected Behaviour
To lint .sql file
## Observed Behaviour
Throws error as quoted.
## Steps to Reproduce
I have 3.9.1 python recently installed.
## Version
Include the output of `sqlfluff --version` along with your Python version
sqlfluff, version 0.3.6
Python 3.9.1
## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = snowflake
templater = jinja
rules = None
exclude_rules = None
recurse = 0
output_line_length = 80

[sqlfluff:indentation]
indented_joins = False

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
generate_schema_name = {% macro generate_schema_name(custom_schema_name, node) -%} blabla {%- endmacro %}

row_limit = {% macro row_limit() -%} limit 100 {%- endmacro %}
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 160
indent_unit = space

# Some rules have their own specific config.
[sqlfluff:rules:L010]
capitalisation_policy = consistent

[sqlfluff:rules:L019]
comma_style = trailing
```
It is very strange behaviour because some files such as this give this error while when I lint a file in a different directory I get this:
```
sqlfluff lint models/1_RAW/MP_MASTER_EVENT.sql                   
== [models/1_RAW/MP_MASTER_EVENT.sql] FAIL
L:   2 | P:   1 | L001 | Unneccessary trailing whitespace.
L:   4 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:   4 | P:   2 |  PRS | Found unparsable section: 'metadata$filename as
                       | FILE_NAME'
L:   4 | P:   2 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
L:   4 | P:  10 |  LXR | Unable to lex characters: ''$filename '...'
L:   5 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:   5 | P:  10 |  PRS | Found unparsable section:
                       | 'SUBSTR(metadata$filename,32,10)'
L:   5 | P:  10 | L030 | Inconsistent capitalisation of function names.
L:   5 | P:  25 |  LXR | Unable to lex characters: ''$filename,'...'
L:   5 | P:  35 | L008 | Commas should be followed by a single whitespace unless
                       | followed by a comment.
L:   5 | P:  38 | L008 | Commas should be followed by a single whitespace unless
                       | followed by a comment.
L:   5 | P:  42 | L008 | Commas should be followed by a single whitespace unless
                       | followed by a comment.
L:   5 | P:  56 | L010 | Inconsistent capitalisation of keywords.
L:   6 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:   6 | P:   2 |  LXR | Unable to lex characters: ''$1 as PAYL'...'
L:   6 | P:   2 |  PRS | Found unparsable section: '$1 as PAYLOAD'
L:   6 | P:   2 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
L:   7 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:   7 | P:   2 | L030 | Inconsistent capitalisation of function names.
L:   7 | P:   7 | L014 | Inconsistent capitalisation of unquoted identifiers.
L:   7 | P:  29 | L008 | Commas should be followed by a single whitespace unless
                       | followed by a comment.
L:   7 | P:  34 | L010 | Inconsistent capitalisation of keywords.
L:   8 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:   8 | P:   3 | L003 | Indentation not consistent with line #7
L:   8 | P:   3 | L030 | Inconsistent capitalisation of function names.
L:   8 | P:  56 | L010 | Inconsistent capitalisation of keywords.
L:  10 | P:   1 |  PRS | Found unparsable section: 'from\n
                       | @s3_mixpanel_mp_master_event'
L:  10 | P:   1 | L010 | Inconsistent capitalisation of keywords.
L:  11 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  11 | P:   3 |  LXR | Unable to lex characters: ''@s3_mixpan'...'
L:  13 | P:   1 | L010 | Inconsistent capitalisation of keywords.
L:  13 | P:  21 | L010 | Inconsistent capitalisation of keywords.
L:  14 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  14 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  14 | P:  20 | L001 | Unneccessary trailing whitespace.
L:  15 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  15 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  15 | P:  31 | L001 | Unneccessary trailing whitespace.
L:  16 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  16 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  16 | P:  18 | L001 | Unneccessary trailing whitespace.
L:  17 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  17 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  17 | P:  17 | L001 | Unneccessary trailing whitespace.
L:  18 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  18 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  19 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  19 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  19 | P:  17 | L001 | Unneccessary trailing whitespace.
L:  20 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  20 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  20 | P:  15 | L001 | Unneccessary trailing whitespace.
L:  21 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  21 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  22 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  22 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  23 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  23 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  24 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  24 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  25 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  25 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  26 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  26 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  27 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  27 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  28 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  28 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  29 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  29 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  30 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  30 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  31 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  31 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  32 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  32 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  33 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  33 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  34 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  34 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  35 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  35 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  35 | P:   4 | L001 | Unneccessary trailing whitespace.
L:  37 | P:   1 | L001 | Unneccessary trailing whitespace.
L:  37 | P:   1 | L004 | Mixed Tab and Space indentation found in file.
L:  40 | P:   1 | L010 | Inconsistent capitalisation of keywords.
L:  40 | P:   7 | L009 | Files must end with a trailing newline.
```

**Comment by drewmcdonald** [comment]
I've added a more tightly scoped example to #661. Hope that helps!

**Comment by MinchoG** [comment]
I have removed all whitespaces on macros and continue to receive such an error, can we discuss it privately to show code snippet and consider options to solve this, please?

**Comment by drewmcdonald** [comment]
@MinchoG this is fixed in #689 ; will defer to the sqlfluff folks on when it'll be released

**Comment by NiallRees** [comment]
This is now released in 0.4.0 :)
